src/hooks.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2019-2020, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useCallback, useContext, useEffect, useState } from 'react';
+import * as optimizely from '@optimizely/optimizely-sdk';
+import { getLogger } from '@optimizely/js-sdk-logging';
+
+import { setupAutoUpdateListeners } from './autoUpdate';
+import { VariableValuesObject, OnReadyResult } from './client'
+import { OptimizelyContext } from './Context';
+
+type UseFeatureState = {
+  isEnabled: Boolean,
+  variables: VariableValuesObject,
+};
+
+type ClientReady = Boolean;
```

Comment:
Does declaring a new type like this gain us anything? Readability/documentation perhaps? In other languages I've seen examples like this where you can leverage the type system to avoid mixing up primitive values representing different things in your domain. But I don't know if TS can be used this way or if that would be useful here.

Reply:
So I did this to document specifically the return interface of `useFeature` (see `UseFeature` below).

I'm totally open to other ideas here, but since the convention of Hooks is to return a tuple like list, I wanted the types to be self documenting about what you get back.

E.g.:

```js
[
    UseFeatureState["isEnabled"],
    UseFeatureState["variables"],
    ClientReady
]
```

instead of just 

```js
[
    UseFeatureState["isEnabled"],
    UseFeatureState["variables"],
    boolean
]
```

Reply:
I see the gain in readability 👍